### PR TITLE
[brightcove] add support for brightcove in page embed(fixes #6824)(fixes #5946)

### LINKED ISSUE #5946: Geek and Sundry
I don't have time to further look into this myself right now, but here's some basic info I gathered for future reference.

Regular page URL:
http://geekandsundry.com/titansgrave-chapter-0/

Relevant HTML:
```html
    <div class="video-wrapper">
        <div class="video-inner-wrapper container">
            
                        <video
                id="video_4279049078001"
                data-account="4036320279001"
                data-player="5d112ed9-283f-485f-a7f9-33f42e8bc042"
                data-embed="default"
                data-video-id="4279049078001"
                class="video-js"
                autoplay="true"
                controls>
            </video>
```

Relevant streaming playlist link used by the Flash player (`videoId` is `data-video-id` and `pubId` is `data-account`): `http://c.brightcove.com/services/mobile/streaming/index/master.m3u8?videoId=4279049078001&pubId=4036320279001`

### LINKED ISSUE #6824: brightcove other URL type support
http://www.dn.no/magasinet/2015/09/07/1606/Popcorn-Time/inside-popcorn-time--the-worlds-fastest-growing-piracy-site

### PR DESCRIPTION
**Pull request by remitamine**
the same error happen here again (https://github.com/rg3/youtube-dl/pull/6785#issue-105191077) if someone has access to the site and can run the tests and check if they need to be changed

### CODE REVIEW

**Comment by remitamine** [comment]
as i said:
> cannot download extracted videos urls
> i can only access the json request
so i just guess that urls like that(streaming_src in api json output):
ttp://brightcove06-f.akamaihd.net/929656772001/201509/,1022/929656772001_4463689346,1238/929656772001_4463679919,2898/929656772001_4463667480,3238/929656772001_4463679901,3898/929656772001_4463667481,3898/929656772001_4463667493,898/929656772001_4463667510,001_Popcorn-Time--English.mp4.csmil/bitrate=2?__nn__=795285514001&primaryToken=1442406600_89d3546e357d8b62ebff54243b5f8f18&pubId=929656772001&videoId=4463358922001
are f4m but as you get an error i think they are smil manifests that link to other links(like in theplatform).
if you can confirm what is in the file or send the file content so i can fix it.

**Comment by remitamine** [comment]
the NerdistIE can be removed because they use the brightcove in page embed

**Comment by remitamine** [comment]
it should work now.

**Comment by remitamine** [comment]
i fixed some mistakes i made before(copy and paste) in the test and field extraction.

File: youtube_dl/utils.py [hunk 1]
```diff
@@ -248,6 +248,14 @@ def get_element_by_attribute(attribute, value, html):
     return unescapeHTML(res)
 
 
+def extract_attributes(attributes_str, attributes_regex=r'(?s)\s*([^\s=]+)\s*=\s*["\']([^"\']+)["\']'):
+    attributes = re.findall(attributes_regex, attributes_str)
+    attributes_dict = {}
+    if attributes:
+        attributes_dict = {attribute_name: attribute_value for (attribute_name, attribute_value) in attributes}
```
Comment (dstftw) [inline youtube_dl/utils.py]: dict comprehensions unsupported in python 2.6.

Comment (remitamine) [inline youtube_dl/utils.py]: the code is the same in #6811.
i just need an answer for https://github.com/rg3/youtube-dl/pull/6811#issuecomment-139608906.
and then i will fix it and keep it in single place.

Comment (dstftw) [inline youtube_dl/utils.py]: Include it here.

**Comment by remitamine** [comment]
i changed the code it should work in python 2.6.

**Comment by dstftw** [comment]
Thanks.

**Pull request merged**